truffleruby/src/main/java/org/truffleruby/core/binding/BindingNodes.java

```diff
@@ -96,12 +61,42 @@ public static FrameDescriptor newFrameDescriptor(RubyContext context) {
 
     public static FrameDescriptor getFrameDescriptor(DynamicObject binding) {
         assert RubyGuards.isRubyBinding(binding);
-        return Layouts.BINDING.getFrame(binding).getFrameDescriptor();
+        return getDeclarationFrame(binding).getFrameDescriptor();
     }
 
     public static MaterializedFrame getDeclarationFrame(DynamicObject binding) {
```

Comment:
This needs a better name.
"declaration frame" means the parent lexical frame of a frame.
Maybe `getTopFrame()`?